src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -481,7 +482,21 @@ private String getNameFromType(String type, boolean isArg) {
 			} else {
 				int count = nameCounts.compute(varName, (k, v) -> (v == null) ? 1 : v + 1);
 
-				return count == 1 ? varName : varName.concat(Integer.toString(count));
+				if (count > 1) {
+					varName = varName.concat(Integer.toString(count));
+				}
+				/*
+				 * Perform keyword check here, because appending numbers for duplicate names
+				 * should prevent keywords, so result will be:
+				 *   class_, class2, class3
+				 * If keyword check was performed before checking for duplicates, result would be:
+				 *   classs_, class_2, class_3
+				 */
+				else if (getInvalidJavaIdentifierReason(varName) == InvalidIdentifierReason.KEYWORD) {
```

Comment:
I'd use something like `} else {\n/* long\n comment*/ if (getInvalidJavaIdent[...]) {\n[...]\n}\n}` to retain the formatting style.

Reply:
Thanks, that comment does not exist in the new version anymore